lib/axon.ex

```diff
@@ -82,6 +82,11 @@ defmodule Axon do
   @type t :: %__MODULE__{}
 
   @doc false
+  @derive {
+    Nx.Container,
+    containers: [],
+    keep: [:id, :name, :output_shape, :parent, :op, :params, :policy, :opts]
```

Comment:
Do you need to access all of those fields inside `defn`? If so, it is fine to keep. If not, they shouldn't be really listed here. If there is a field you want to return as a result of `defn`, then you can wrap it with the original struct on the way out:

    {defn_struct, ....} = call_defn(my_struct, ...)
    Map.merge(my_struct, Map.take(defn_struct, container_fields))